### ISSUE TITLE
Bad results when testing

### ISSUE THREAD

**Issue by ShaniGam** [NVIDIA/pix2pixHD#162]
I train the model on 2975 images from the cityscapes dataset with the following command:
`python train.py --name label2city_512p_feat_cityscapes --instance_feat`
The outputs I get during the training seem fine:
![epoch054_160500_synthesized_image](https://user-images.githubusercontent.com/11832259/68186394-b523f080-ffac-11e9-876b-5b44c5400f18.jpg)
![epoch055_161000_synthesized_image](https://user-images.githubusercontent.com/11832259/68186395-b523f080-ffac-11e9-809f-0d58f95d06a7.jpg)

But when I test the model (on images from the training set) I get:

![aachen_000004_000019_gtFine_labelIds_synthesized_image](https://user-images.githubusercontent.com/11832259/68186811-b1449e00-ffad-11e9-8266-f58a828d0bb2.jpg)
![aachen_000010_000019_gtFine_labelIds_synthesized_image](https://user-images.githubusercontent.com/11832259/68186812-b1449e00-ffad-11e9-8a03-4edc0747f131.jpg)

Does anyone has an idea of why it happens?

**Comment by cszer** [comment]
overfitting

**Comment by wang-zm18** [comment]
Did you solve the problem? I also meet with this problem during testing.

**Comment by yxwang1794** [comment]
can we add a validation set to prevent overfitting in pix2pixHD model? I haven't seen anyone
 done this before.

**Comment by edd2110-jac** [comment]
> overfitting

I don't think it's because overfitting. He says that he is "Testing the model with images from the training set". If it is overfitting. It shouldn't work with any other images but with images the model was trained with should work. 

I'm having this same issue, with no good outcome yet.

**Comment by songyn95** [comment]
How did you fix it? @ShaniGam @wang-zm18 @cszer @yxwang1794 @edd2110-jac

**Comment by PradeepYLN** [comment]
@ShaniGam @wang-zm18  @cszer  @yxwang1794  @edd2110-jac @songyn95 
Has anyone added the validation step to check for overfitting?
